lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp

```diff
@@ -6031,6 +6056,80 @@ CompilerType TypeSystemClang::GetVirtualBaseClassAtIndex(
   return CompilerType();
 }
 
+std::optional<int64_t>
+TypeSystemClang::TryToGetBaseOffset(const clang::CXXRecordDecl *derived,
+                                    const clang::CXXRecordDecl *base,
+                                    clang::CXXBasePaths *paths) {
+  bool is_ambiguous = paths->isAmbiguous(base->getTypeForDecl()
+                                             ->getCanonicalTypeInternal()
+                                             ->getCanonicalTypeUnqualified());
+  if (is_ambiguous)
+    return std::nullopt;
+
+  int64_t offset = 0;
+
+  for (auto path_elem : llvm::reverse(paths->front())) {
+    const clang::CXXRecordDecl *base_decl =
+        path_elem.Base->getType()->getAsCXXRecordDecl();
```

Comment:
`getAsCXXRecordDecl()` theoretically can return `nullptr`. It wouldn't make much sense, because this entity has to be able to participate in inheritance. You should assert that `base_decl` is not `nullptr`.